### PR TITLE
Introduce React.ElementConfig<> to mimic the same concept from Flow

### PR DESCRIPTION
**Pull request by jeremywiebe**
## Summary:

TypeScript defines a very verbose `JSX.LibraryManagedAttributes<T, Props>` for use when you want to get the type of a components Props with the optionality of those props maintained (per default props). 

This is cumbersome to type and promote in usage and so this PR introduces a "global ambient" type on the `React` namespace named `ElementConfig`. We've chosen this name because it's the exact same name as Flow has for this same concept (thanks @somewhatabstract). 

:warning: Note: there are three ESLint errors from the `react/prop-types` rule that we need to resolve before landing this PR.

Issue: "none"

## Test plan:

`yarn typecheck` is happy

### CODE REVIEW

File: packages/perseus/src/hints-renderer.tsx [hunk 1]
```diff
@@ -195,10 +189,8 @@ class HintsRenderer extends React.Component<Props, State> {
                     ref={"hintRenderer" + i}
                     key={"hintRenderer" + i}
                     apiOptions={apiOptions}
-                    // @ts-expect-error [FEI-5003] - TS2339 - Property 'findExternalWidgets' does not exist on type 'Readonly<{ className?: string | undefined; hints: readonly any[]; hintsVisible?: number | undefined; }> & Readonly<{ children?: ReactNode; }>'.
```
Comment (jeremywiebe) [inline packages/perseus/src/hints-renderer.tsx:198]: Typescript is now fine with this line, but ESLint's `react/prop-types` rule complains with: `'findExternalWidgets' is missing in props validation` [react/prop-types](https://github.com/jsx-eslint/eslint-plugin-react/tree/master/docs/rules/prop-types.md)

Comment (kevinbarabash) [reply packages/perseus/src/hints-renderer.tsx:198]: I think we can get rid of that rule since we're using TS for prop type checking.

**Comment by jeremywiebe** [comment]
Right now this is blocked because `yarn build:types` fails (it doesn't seem to see the custom `ElementConfig<T>` type that are defined in `types/react.d.ts`).

**Comment by somewhatabstract** [comment]
@jeremywiebe I wonder if we should export this type from Wonder Stuff Core?

**Comment by jeremywiebe** [comment]
> @jeremywiebe I wonder if we should export this type from Wonder Stuff Core? After all, we will want it in webapp too. We can call it something else like `PropsFrom`.

I would LOVE that!  I could never find a way to get `build:types` to work the way I was trying to set it up.

**Comment by somewhatabstract** [comment]
> > @jeremywiebe I wonder if we should export this type from Wonder Stuff Core? After all, we will want it in webapp too. We can call it something else like `PropsFrom`.
> 
> I would LOVE that! I could never find a way to get `build:types` to work the way I was trying to set it up.

OK, I will see if I can do that right now :)

**Comment by somewhatabstract** [comment]
@jeremywiebe https://github.com/Khan/wonder-blocks/pull/1980

**Comment by somewhatabstract** [comment]
@jeremywiebe the latest WB Core has this type now. I'm just writing a blog with the info.

**Pull request closed without merging**